gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/DevicesResource.java

```diff
@@ -78,7 +79,7 @@ public void list(
         checkAnyPermission(organizationId, environmentId, domain, Permission.DOMAIN_USER_DEVICE, Acl.LIST)
                 .andThen(domainService.findById(domain)
                         .switchIfEmpty(Maybe.error(new DomainNotFoundException(domain)))
-                        .flatMapSingle(__ -> this.deviceService.findByDomainAndUser(domain, user).toList()))
+                        .flatMapSingle(__ -> this.deviceService.findByDomainAndUser(domain, UserId.internal(user)).toList()))
```

Comment:
**todo**: we need to search by (internalId or (externalId and source)) to be sure to retrieve all the devices for a single user. So you have to search the user first in the same way as we do for the domain to build the fullUserId.